osu.Game/Screens/OnlinePlay/Lounge/Components/DrawableRoom.cs

```diff
@@ -224,8 +242,64 @@ protected override void LoadComplete()
                 Alpha = 0;
         }
 
+        public Popover GetPopover() => new PasswordEntryPopover(Room);
+
+        public MenuItem[] ContextMenuItems => new MenuItem[]
+        {
+            new OsuMenuItem("Create copy", MenuItemType.Standard, () =>
+            {
+                parentScreen?.OpenNewRoom(Room.CreateCopy());
+            })
+        };
+
+        public bool OnPressed(GlobalAction action)
+        {
+            if (selectedRoom.Value != Room)
+                return false;
+
+            switch (action)
+            {
+                case GlobalAction.Select:
+                    Click();
+                    return true;
+            }
+
+            return false;
+        }
+
+        public void OnReleased(GlobalAction action)
+        {
+        }
+
         protected override bool ShouldBeConsideredForInput(Drawable child) => state == SelectionState.Selected;
 
+        protected override bool OnMouseDown(MouseDownEvent e)
+        {
+            if (selectedRoom.Value != Room)
+                return true;
```

Comment:
What's the purpose of this?

Reply:
This is so the global action directly below doesn't run when the room is not selected.